src/ProjectFileTools.MSBuild/Worksapce.cs

```diff
@@ -0,0 +1,149 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using Microsoft.Build.Construction;
+using Microsoft.Build.Evaluation;
+using Microsoft.Language.Xml;
+
+namespace ProjectFileTools.MSBuild
+{
+    /// <summary>
+    /// Contains an MSBuild project and logic to extract information from it
+    /// </summary>
+    public class Workspace
+    {
+        private ProjectCollection _collection;
+        private Project _project;
+        private HashSet<string> _containedFiles;
+        private List<FileSystemWatcher> _watchers;
+        private bool _needsReload;
+
+        internal Workspace(string filePath)
+        {
+            _collection = new ProjectCollection();
+            _containedFiles = new HashSet<string>(StringComparer.Ordinal);
+            _watchers = new List<FileSystemWatcher>();
+            _needsReload = false;
+
+            try
+            {
+                _project = _collection.LoadProject(filePath);
+                UpdateContainedFiles();
+            }
+            // TODO: Propagate error to the errors list
+            catch
+            {
+                _project = null;
+            }
+        }
+
+        /// <summary>
+        /// Returns the URL of the file that contains the definition of the item at the current position 
+        /// </summary>
+        /// <param name="filePath">Current file</param>
+        /// <param name="sourceText">Text in the current file</param>
+        /// <param name="position">Position of item that is to be resolved</param>
+        /// <returns></returns>
+        public string ResolveDefinition(string filePath, string sourceText, int position)
+        {
+            String file = null;
+
+            if (_project != null)
+            {
+                XmlDocumentSyntax root = Parser.ParseText(sourceText);
+                SyntaxNode syntaxNode = SyntaxLocator.FindNode(root, position);
+
+                if (syntaxNode.ParentElement.Name.Equals("Import"))
```

Comment:
can any part of this call chain be null? If the user f12's on a root node, will this throw?